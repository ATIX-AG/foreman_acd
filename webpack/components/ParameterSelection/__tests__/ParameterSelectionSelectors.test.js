import { testSelectorsSnapshotWithFixtures } from 'react-redux-test-utils';

import {
  selectLoading,
  selectEditMode,
  selectParamData,
  selectAllowedParameterTypes,
  selectParameterTypes,
  selectParameters,
  selectSortingColumns,
  selectColumns,
  selectParamDefinition,
  selectHostgroupId,
  selectEditParamsRowIndex,
} from '../ParameterSelectionSelectors';

import { parameterSelectionData1 } from '../__fixtures__/parameterSelectionData1.fixtures';

const stateFactory = obj => ({
  foremanAcd: {
    parameterSelectionParameters: obj,
  },
});

const fixtures = {
  'should return loading from parameterSelectionData1 fixtures': () =>
    selectLoading(stateFactory(parameterSelectionData1)),
  'should return editMode from parameterSelectionData1 fixtures': () =>
    selectEditMode(stateFactory(parameterSelectionData1)),
  'should return paramData from parameterSelectionData1 fixtures': () =>
    selectParamData(stateFactory(parameterSelectionData1)),
  'should return allowedParameterTypes from parameterSelectionData1 fixtures': () =>
    selectAllowedParameterTypes(stateFactory(parameterSelectionData1)),
  'should return parameterTypes from parameterSelectionData1 fixtures': () =>
    selectParameterTypes(stateFactory(parameterSelectionData1)),
  'should return parameters from parameterSelectionData1 fixtures': () =>
    selectParameters(stateFactory(parameterSelectionData1)),
  'should return sortingColumns from parameterSelectionData1 fixtures': () =>
    selectSortingColumns(stateFactory(parameterSelectionData1)),
  'should return columns from parameterSelectionData1 fixtures': () =>
    selectColumns(stateFactory(parameterSelectionData1)),
  'should return ParamDefinition from parameterSelectionData1 fixtures': () =>
    selectParamDefinition(stateFactory(parameterSelectionData1)),
  'should return hostgroupId from parameterSelectionData1 fixtures': () =>
    selectHostgroupId(stateFactory(parameterSelectionData1)),
  'should return editParamsRowIndex from parameterSelectionData1 fixtures': () =>
    selectEditParamsRowIndex(stateFactory(parameterSelectionData1)),
};

describe('ParameterSelectionSelectors', () =>
  testSelectorsSnapshotWithFixtures(fixtures));
