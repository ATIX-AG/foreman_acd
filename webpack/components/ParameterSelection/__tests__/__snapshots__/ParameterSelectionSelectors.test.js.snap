// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParameterSelectionSelectors should return ParamDefinition from parameterSelectionData_1 fixtures 1`] = `
Object {
  "dataId": "1",
  "id": 1,
  "name": "web",
}
`;

exports[`ParameterSelectionSelectors should return columns from parameterSelectionData_1 fixtures 1`] = `
Array [
  Object {
    "cell": Object {
      "formatters": Array [
        null,
      ],
    },
    "header": Object {
      "customFormatters": Array [
        null,
      ],
      "formatters": Array [
        null,
      ],
      "label": "Name",
      "props": Object {
        "index": 0,
        "sort": true,
        "style": Object {
          "width": "25%",
        },
      },
      "transforms": Array [
        null,
      ],
    },
    "property": "name",
  },
  Object {
    "cell": Object {
      "formatters": Array [
        null,
      ],
      "props": Object {
        "index": 1,
      },
    },
    "header": Object {
      "customFormatters": Array [
        null,
      ],
      "formatters": Array [
        null,
      ],
      "label": "Description",
      "props": Object {
        "index": 1,
        "sort": true,
        "style": Object {
          "width": "25%",
        },
      },
      "transforms": Array [
        null,
      ],
    },
    "property": "description",
  },
  Object {
    "cell": Object {
      "formatters": Array [
        null,
      ],
    },
    "header": Object {
      "customFormatters": Array [
        null,
      ],
      "formatters": Array [
        null,
      ],
      "label": "Type",
      "props": Object {
        "index": 2,
        "sort": true,
        "style": Object {
          "width": "20%",
        },
      },
      "transforms": Array [
        null,
      ],
    },
    "property": "type",
  },
  Object {
    "cell": Object {
      "formatters": Array [
        null,
      ],
    },
    "header": Object {
      "customFormatters": Array [
        null,
      ],
      "formatters": Array [
        null,
      ],
      "label": "Default value",
      "props": Object {
        "index": 3,
        "sort": true,
        "style": Object {
          "width": "20%",
        },
      },
      "transforms": Array [
        null,
      ],
    },
    "property": "value",
  },
  Object {
    "cell": Object {
      "formatters": Array [
        null,
      ],
    },
    "header": Object {
      "formatters": Array [
        null,
      ],
      "label": "Actions",
      "props": Object {
        "index": 4,
        "style": Object {
          "width": "10%",
        },
      },
    },
    "property": "actions",
  },
]
`;

exports[`ParameterSelectionSelectors should return editMode from parameterSelectionData_1 fixtures 1`] = `false`;

exports[`ParameterSelectionSelectors should return hostgroupId from parameterSelectionData_1 fixtures 1`] = `1`;

exports[`ParameterSelectionSelectors should return loading from parameterSelectionData_1 fixtures 1`] = `false`;

exports[`ParameterSelectionSelectors should return paramData from parameterSelectionData_1 fixtures 1`] = `
Object {
  "computeprofiles": Array [
    Object {
      "id": 1,
      "name": "1-Small",
    },
    Object {
      "id": 2,
      "name": "2-Medium",
    },
    Object {
      "id": 3,
      "name": "3-Large",
    },
    Object {
      "id": 4,
      "name": "Orchahosts-VM",
    },
  ],
  "dataType": "PARAMETER_SELECTION_PARAM_TYPE_FOREMAN",
  "domains": Array [
    Object {
      "id": 1,
      "name": "deploy3.dev.atix",
    },
  ],
  "environments": Array [
    Object {
      "id": 1,
      "name": "production",
    },
  ],
  "hostgroup_id": 1,
  "lifecycle_environments": Array [
    Object {
      "id": 1,
      "name": "Library",
    },
  ],
  "ptables": Array [
    Object {
      "id": 125,
      "name": "Kickstart default",
    },
  ],
}
`;

exports[`ParameterSelectionSelectors should return parameterTypes from parameterSelectionData_1 fixtures 1`] = `
Object {
  "domain": "Domain",
  "hostparam": "Host parameter",
  "ip": "IP",
  "password": "Root password",
  "ptable": "Partition table",
  "puppetenv": "Puppet environment",
}
`;

exports[`ParameterSelectionSelectors should return parameters from parameterSelectionData_1 fixtures 1`] = `
Array [
  Object {
    "description": "",
    "id": 1,
    "locked": false,
    "name": "CP",
    "type": "computeprofile",
    "value": "1",
  },
  Object {
    "description": "",
    "id": 2,
    "locked": true,
    "name": "LE",
    "type": "lifecycleenv",
    "value": "1",
  },
]
`;

exports[`ParameterSelectionSelectors should return sortingColumns from parameterSelectionData_1 fixtures 1`] = `
Object {
  "name": Object {
    "direction": "asc",
    "position": 0,
  },
}
`;
